composer/profiler/profiler.py

```diff
@@ -450,9 +464,9 @@ def start(self) -> None:
             raise RuntimeError(
                 f"Attempted to start profiler event {self.name}; however, this marker is already started")
 
-        epoch = self.profiler.state.epoch
-        step = self.profiler.state.step
-        batch_idx = self.profiler.state.batch_idx
+        epoch = self.state.epoch
+        step = self.state.step
+        batch_idx = self.state.batch_idx
```

Comment:
`state.batch_idx` is deprecated

```suggestion
        batch_idx = int(self.state.timer.batch_in_epoch)
```